Missing types in 8.0.0-beta

## Describe the bug
I tried out the new beta release that integrates with Next's new i18n internationalization features. It works all very nicely, thanks for the fast development towards this new release. There is only one thing that holds me back from deploying it Vercel. 
When importing `{appWithTranslation}` or `{useTranslation}` from 'next-i18next' in any typescript file (e.g. `_app.tsx`) I get the warning `'"../node_modules/next-i18next/types"' has no exported member named 'appWithTranslation'. Did you mean 'AppWithTranslation'?` 
It seems like there are missing exports in that `types.d.ts`. While the local server still works, the build process to Vercel fails.

## Occurs in next-i18next version
8.0.0-beta.0

## Steps to reproduce
Clone the demo repo at https://github.com/isaachinman/next-i18next-vercel and simply change the file extension of `pages/_app.js` to `pages/_app.tsx`. The import warning will then be displayed.

## Expected behaviour
Correct type definitions.

I'm not sure this is related but as I see the same issues in addition to that when I go to page with `useTranslation` component I'm seeing this error: 
```error - ./node_modules/next-i18next/dist/commonjs/serverSideTranslations.js:28:0
Module not found: Can't resolve 'fs'
null

The component is exporting `getStaticProps` the same way as in basic example.

--- a/package.json
+++ b/package.json
@@ -35,7 +35,7 @@
     "build:examples/simple": "yarn --cwd examples/simple && yarn --cwd examples/simple build",
     "prepublishOnly": "yarn build",
     "run-example": "yarn build && cd examples/simple && yarn && yarn dev",
-    "run-example:prod": "yarn --cwd examples/simple start",
+    "run-example:prod": "yarn build:examples/simple && yarn --cwd examples/simple start",
     "test": "yarn check-types && yarn clean && yarn build:cjs && yarn build:examples/simple && bundlesize && NODE_ENV=test jest --passWithNoTests --maxWorkers=1 --silent",
     "contributors:check": "all-contributors check",
     "contributors:add": "all-contributors add",

--- a/src/appWithTranslation.tsx
+++ b/src/appWithTranslation.tsx
@@ -12,7 +12,7 @@ type AppProps = {
   pageProps: SSRConfig;
 }
 
-export const appWithTranslation = (WrappedComponent: React.ComponentType): React.ComponentType => {
+export const appWithTranslation = <P extends Record<string, unknown>>(WrappedComponent: React.ComponentType | React.ElementType): React.ComponentType<P> | React.ElementType<P> => {
   const AppWithTranslation = (props: AppProps) => {
     let i18n = null
     let locale = null

--- a/types.d.ts
+++ b/types.d.ts
@@ -9,6 +9,7 @@ import {
   WithTranslation as ReactI18nextWithTranslation
 } from 'react-i18next'
 import { InitOptions, i18n, TFunction as I18NextTFunction } from 'i18next'
+import { appWithTranslation } from './src';
 
 export type InitConfig = {
   strictMode?: boolean;
@@ -38,7 +39,7 @@ export type NextI18NextInternals = {
 
 export type Trans = (props: TransProps) => any
 export type UseTranslation = typeof useTranslation
-export type AppWithTranslation = <P extends unknown>(Component: React.ComponentType<P> | React.ElementType<P>) => any
+export type AppWithTranslation = typeof appWithTranslation
 export type TFunction = I18NextTFunction
 export type I18n = i18n
 export type WithTranslationHocType = typeof withTranslation
@@ -59,6 +60,7 @@ export type SSRConfig = {
 
 export {
   I18nContext,
+  appWithTranslation,
   useTranslation,
   withTranslation,
 }

--- a/types.d.ts
+++ b/types.d.ts
@@ -64,16 +64,3 @@ export {
   useTranslation,
   withTranslation,
 }
-
-declare class NextI18Next {
-  constructor(config: InitConfig);
-  Trans: Trans
-  i18n: I18n
-  initPromise: InitPromise
-  config: Config
-  useTranslation: UseTranslation
-  withTranslation: WithTranslationHocType
-  appWithTranslation: AppWithTranslation
-}
-
-export default NextI18Next

